### PR TITLE
try removing webpacker lines from GitPod config

### PR DESCRIPTION
**Pull request by jywarren**
re #9502

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ tasks:
 
         yarn install
 
-        rails g webpacker:install && rails g webpacker:install:react && rails g react:install
+        rails webpacker:install && rails webpacker:install:react && rails g react:install
```
Comment (jywarren) [inline .gitpod.yml:21]:
```suggestion
```

Comment (jywarren) [reply .gitpod.yml:21]: OK, i see this message in the output in GitPod:

```
    conflict  config/webpacker.yml
Overwrite /workspace/plots2/config/webpacker.yml? (enter "h" for help) [Ynaqdhm]
```

We could decide to move fwd with removing that entire line, and see what it does in GitPod?

**Comment by jywarren** [comment]
hmm, now says:

```

ActionView::Template::Error (Webpacker can't find application.js in /workspace/plots2/public/packs/manifest.json. Possible causes:
1. You want to set webpacker.yml value of compile to true for your environment
   unless you are using the `webpack -w` or the webpack-dev-server.
2. webpack has not yet re-run to reflect updates.
3. You have misconfigured Webpacker's config/webpacker.yml file.
4. Your webpack configuration is not creating a manifest.
Your manifest contains:
{
}
):
    107: 
    108:   <script defer src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-slider/10.2.0/bootstrap-slider.min.js"></script>
    109:   <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-slider/10.2.0/css/bootstrap-slider.min.css" />
    110:   <%= javascript_pack_tag 'application' %>
    111:   </head>
    112: 
```

**Comment by noi5e** [comment]
The `manifest.json` error @jywarren posted about appears in the GitPod:
Screen Shot 2021-09-07 at 10 51 40 AM

Could this be applicable here?
https://github.com/publiclab/plots2/blob/d35d93520a8097ca70d6b1d446e6f942873d00e3/config/webpacker.yml#L83-L87

Does this count as a `production` environment? If so, maybe change this variable to `true`? 🤔

File: .gitpod.yml [hunk 2]
```diff
@@ -18,7 +18,6 @@ tasks:
 
         yarn install
 
-        rails g webpacker:install && rails g webpacker:install:react && rails g react:install
```
Comment (jywarren) [inline .gitpod.yml:21]: I think i would guess it's `2) webpack has not yet re-run to reflect updates` -- so, if we remove the `g` lines, which of these three ought we to keep if any?

`rails webpacker:install && rails webpacker:install:react && rails g react:install`

**Comment by jywarren** [comment]
OK, building on notes in #9502, let's try linking each line to where we find what it does, in upstream source:

- [ ] `rails webpacker:install`
- [ ] `rails webpacker:install:react`
- [ ] `rails g react:install`

**Comment by jywarren** [comment]
> Does this count as a `production` environment? If so, maybe change this variable to `true`? 🤔

I think not. I think if you turn it on, it'll try to on-the-fly compile in production. But we want that only to happen in development. And GitPod would be running in development mode unless we explicitly switched it.

**Comment by jywarren** [comment]
Interesting possibly related error: https://github.com/rails/webpacker/issues/1523 (but for vue)

**Comment by jywarren** [comment]
Wait, ok, so on reading[ the error above once more](https://github.com/publiclab/plots2/pull/10137#issuecomment-914505897), I wonder @noi5e do we need to check in manifest.json to version tracking?

**Comment by jywarren** [comment]
Could this be relevant? 

https://github.com/rails/webpacker/blob/79c7b677ed20a650d45776ed173c7a5ecd6c4c6d/docs/troubleshooting.md#cant-find-hello_reactjs-in-manifestjson

> If you get this error Can't find hello_react.js in manifest.json when loading a view in the browser it's because webpack is still compiling packs. Webpacker uses a manifest.json file to keep track of packs in all environments, however since this file is generated after packs are compiled by webpack. So, if you load a view in browser whilst webpack is compiling you will get this error. Therefore, make sure webpack (i.e ./bin/webpack-dev-server) is running and has completed the compilation successfully before loading a view.

**Comment by jywarren** [comment]
more discussion in https://github.com/rails/webpacker/issues/2071 too... for context...

**Comment by jywarren** [comment]
OK, tried (in gitpod) running just `rails g react:install`:

```
gitpod /workspace/plots2 $ rails g react:install
W, [2021-09-14T18:37:02.203538 #7060]  WARN -- Skylight: [SKYLIGHT] [5.1.1] Running Skylight in development mode. No data will be reported until you deploy your app.
(To disable this message for all local apps, run `skylight disable_dev_warning`.)
       exist  app/javascript/components
   identical  app/javascript/components/.keep
warning Pattern ["file-saver@eligrey/FileSaver.js#1.3.8"] is trying to unpack in the same destination "/home/gitpod/.cache/yarn/v6/npm-file-saver-1.3.8/node_modules/file-saver" as pattern ["file-saver@^1.3.3"]. This could result in non-deterministic behavior, skipping.
warning " > @babel/preset-react@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".                       
warning "@babel/preset-react > @babel/plugin-transform-react-display-name@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-jsx@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-jsx-development@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-pure-annotations@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-jsx > @babel/plugin-syntax-jsx@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > leaflet-spin@1.1.0" has incorrect peer dependency "leaflet@^0.7.3".                                 
warning " > webpack-dev-server@3.11.2" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".                    
warning "webpack-dev-server > webpack-dev-middleware@3.7.3" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
warning Ignored scripts due to flag.                                                                            
File unchanged! The supplied flag value not found!  app/javascript/packs/application.js
   identical  app/javascript/packs/server_rendering.js
```

So, i think that's what `webpack-dev-server` is for? To compile this on the fly for development use? 

I tried running it following the docs: https://github.com/webpack/webpack-dev-server#usage

```

gitpod /workspace/plots2 $ webpack-dev-server
bash: webpack-dev-server: command not found
gitpod /workspace/plots2 $ webpack serve
bash: webpack: command not found
gitpod /workspace/plots2 $ bin/webpack serve
The command moved into a separate package: @webpack-cli/serve
Would you like to install serve? (That will run yarn add -D @webpack-cli/serve) (yes/NO) : gitpod /workspace/plots2 $ npm run serve
npm ERR! Invalid version: "2.1"

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/gitpod/.npm/_logs/2021-09-14T18_46_26_222Z-debug.log
gitpod /workspace/plots2 $ bin/webpack serve
The command moved into a separate package: @webpack-cli/serve
Would you like to install serve? (That will run yarn add -D @webpack-cli/serve) (yes/NO) : yes
yarn add v1.22.4
[1/4] Resolving packages...
[2/4] Fetching packages...
warning Pattern ["file-saver@eligrey/FileSaver.js#1.3.8"] is trying to unpack in the same destination "/home/gitpod/.cache/yarn/v6/npm-file-saver-1.3.8/node_modules/file-saver" as pattern ["file-saver@^1.3.3"]. This could result in non-deterministic behavior, skipping.
info fsevents@2.3.2: The platform "linux" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > @babel/preset-react@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-display-name@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-jsx@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-jsx-development@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-pure-annotations@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-jsx > @babel/plugin-syntax-jsx@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > leaflet-spin@1.1.0" has incorrect peer dependency "leaflet@^0.7.3".
warning " > webpack-dev-server@3.11.2" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
warning "webpack-dev-server > webpack-dev-middleware@3.7.3" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
warning " > @webpack-cli/serve@1.5.2" has unmet peer dependency "webpack-cli@4.x.x".
[4/4] Building fresh packages...
warning Ignored scripts due to flag.
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ @webpack-cli/serve@1.5.2
info All dependencies
└─ @webpack-cli/serve@1.5.2
Done in 6.89s.
TypeError: Class constructor ServeCommand cannot be invoked without 'new'
    at runWhenInstalled (/workspace/plots2/node_modules/webpack-cli/bin/utils/prompt-command.js:46:9)
    at /workspace/plots2/node_modules/webpack-cli/bin/utils/prompt-command.js:124:15
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
gitpod /workspace/plots2 $ 
```

**Comment by jywarren** [comment]
OH! But actually the app loaded after `bin/webpack serve` and installing the cli with `yes`. It generated this manifest.json:

```
{
  "application.js": "/packs/js/application-1cd674b55ee7cddfadd4.js",
  "application.js.map": "/packs/js/application-1cd674b55ee7cddfadd4.js.map",
  "entrypoints": {
    "application": {
      "js": [
        "/packs/js/application-1cd674b55ee7cddfadd4.js"
      ],
      "js.map": [
        "/packs/js/application-1cd674b55ee7cddfadd4.js.map"
      ]
    },
    "hello_react": {
      "js": [
        "/packs/js/hello_react-af487bc436f7f27d8ec9.js"
      ],
      "js.map": [
        "/packs/js/hello_react-af487bc436f7f27d8ec9.js.map"
      ]
    },
    "server_rendering": {
      "js": [
        "/packs/js/server_rendering-c2fc005ba852cce0103e.js"
      ],
      "js.map": [
        "/packs/js/server_rendering-c2fc005ba852cce0103e.js.map"
      ]
    }
  },
  "hello_react.js": "/packs/js/hello_react-af487bc436f7f27d8ec9.js",
  "hello_react.js.map": "/packs/js/hello_react-af487bc436f7f27d8ec9.js.map",
  "server_rendering.js": "/packs/js/server_rendering-c2fc005ba852cce0103e.js",
  "server_rendering.js.map": "/packs/js/server_rendering-c2fc005ba852cce0103e.js.map"
}
```

So we need to:

1. ensure the extra package is installed (`The command moved into a separate package: @webpack-cli/serve`), and 
2. run `bin/webpack serve` in gitpod. 
3. think about the workflow for local developers if need be

@noi5e what do you think? I added a commit with the `@webpack-cli/serve` install, and will try adding `bin/webpack serve` to gitpod.yml.

**Comment by jywarren** [comment]
hmm, still got: `Webpacker::Manifest::MissingEntryError in Home#home` for `application.js` - looks like same error. I'm not sure if I'm installing and running `webpacker-dev-server` correctly.

**Comment by jywarren** [comment]
Ah and for reference, i think the code i'm adding may be causing a parallel yarn install? 

```
warning Pattern ["file-saver@eligrey/FileSaver.js#1.3.8"] is trying to unpack in the same destination "/home/runner/.cache/yarn/v6/npm-file-saver-1.3.8/node_modules/file-saver" as pattern ["file-saver@^1.3.3"]. This could result in non-deterministic behavior, skipping.
```

maybe causing the test failures... but need to check more. I just looked at output of functional tests.

**Comment by jywarren** [comment]
A reminder, testing with a forced prebuild is: https://gitpod.io#prebuild/https://github.com/publiclab/plots2/pull/10137/

**Comment by jywarren** [comment]
I think we may need to install webpack-cli itself too.

**Comment by noi5e** [comment]
Late to the party here, but doing (a LOT) of catch-up here... 😅  Reading through everything backwards.

**Comment by noi5e** [comment]
Lots of discussion [here](https://github.com/rails/webpacker/issues/1494), including this comment: https://github.com/rails/webpacker/issues/1494#issuecomment-388305305

The gem `rails/webpacker` is currently on 5.4.3, however, we're running 5.4.2:
https://github.com/publiclab/plots2/blob/42f02dffe46daa0ad31f24689e5f9c9e81eed9b7/yarn.lock#L1152-L1155

I'll make a separate PR and see if bumping the version per that comment above does anything.

**Comment by noi5e** [comment]
Boy this is complicated 🤦. 

I noticed something from [@jywarren's comment 3 days ago](https://github.com/publiclab/plots2/pull/10137#issuecomment-919428290), from the `manifest.json` that was generated:
`"application.js": "/packs/js/application-1cd674b55ee7cddfadd4.js"`

Shouldn't that be looking in the `/packs-test/` directory, since GitPod is a testing environment (I think?)

Here's my local `public/packs-test/manifest.json`:
Screen Shot 2021-09-17 at 7 56 32 PM

**Comment by noi5e** [comment]
I did some upstream hunting, I believe that this is where the `rails g react:install` command is defined:
https://github.com/reactjs/react-rails/blob/master/lib/generators/react/install_generator.rb

**Comment by jywarren** [comment]
> Boy this is complicated 🤦.

ugh, yep!

> I noticed something from [@jywarren's comment 3 days ago](https://github.com/publiclab/plots2/pull/10137#issuecomment-919428290), from the `manifest.json` that was generated:
> `"application.js": "/packs/js/application-1cd674b55ee7cddfadd4.js"`
> 
> Shouldn't that be looking in the `/packs-test/` directory, since GitPod is a testing environment (I think?)

You know, that makes some sense, but I guess i was willing to be OK with how things were given that one of the last few commands i ran seemed to work. What did you think of my theory that `webpack-dev-server` is supposed to generate the compiled files on the fly when running in development mode? If that's the case and we can get it running, maybe we're OK? 

But, did you always run `webpack-dev-server` or `bin/webpack serve` or something similar when you were developing locally?

**Comment by jywarren** [comment]
I'm trying adding this 

```json
  "devDependencies": {
    "@webpack-cli/serve": "^1.5.2",
    "webpack-cli": "^4.8.0",
    "webpack-dev-server": "^4.2.1"
  },
```

**Comment by jywarren** [comment]
I think that may have worked...? that is, after that, i can refresh the preview browser and it loads the site. I even caught the log saying something like `Webpack compiling...` and then printing the following:

```

[Webpacker] Hash: a67f1b540430cd4a0db7
Version: webpack 4.46.0
Time: 5657ms
Built at: 09/20/2021 10:37:23 PM
                                          Asset      Size            Chunks                         Chunk Names
         js/application-1cd674b55ee7cddfadd4.js  1.33 MiB       application  [emitted] [immutable]  application
     js/application-1cd674b55ee7cddfadd4.js.map  1.49 MiB       application  [emitted] [dev]        application
         js/hello_react-af487bc436f7f27d8ec9.js  1.09 MiB       hello_react  [emitted] [immutable]  hello_react
     js/hello_react-af487bc436f7f27d8ec9.js.map  1.26 MiB       hello_react  [emitted] [dev]        hello_react
    js/server_rendering-c2fc005ba852cce0103e.js  1.33 MiB  server_rendering  [emitted] [immutable]  server_rendering
js/server_rendering-c2fc005ba852cce0103e.js.map  1.49 MiB  server_rendering  [emitted] [dev]        server_rendering
                                  manifest.json  1.05 KiB                    [emitted]              
Entrypoint application = js/application-1cd674b55ee7cddfadd4.js js/application-1cd674b55ee7cddfadd4.js.map
Entrypoint hello_react = js/hello_react-af487bc436f7f27d8ec9.js js/hello_react-af487bc436f7f27d8ec9.js.map
Entrypoint server_rendering = js/server_rendering-c2fc005ba852cce0103e.js js/server_rendering-c2fc005ba852cce0103e.js.map
[./app/javascript/components sync recursive ^\.\/.*$] ./app/javascript/components sync ^\.\/.*$ 811 bytes {application} {server_rendering} [built]
[./app/javascript/components/App.js] 4.77 KiB {application} {server_rendering} [optional] [built]
[./app/javascript/components/Comment.js] 4.26 KiB {application} {server_rendering} [built]
[./app/javascript/components/CommentDisplay.js] 1.74 KiB {application} {server_rendering} [built]
[./app/javascript/components/CommentForm.js] 4.08 KiB {application} {server_rendering} [built]
[./app/javascript/components/CommentHeader.js] 2.93 KiB {application} {server_rendering} [built]
[./app/javascript/components/CommentReplies.js] 1.41 KiB {application} {server_rendering} [built]
[./app/javascript/components/CommentToolbar.js] 3.06 KiB {application} {server_rendering} [built]
[./app/javascript/packs/application.js] 1.04 KiB {application} [built]
[./app/javascript/packs/hello_react.jsx] 1.05 KiB {hello_react} [built]
[./app/javascript/packs/server_rendering.js] 301 bytes {server_rendering} [built]
[./node_modules/prop-types/index.js] 707 bytes {application} {hello_react} {server_rendering} [built]
[./node_modules/react-dom/index.js] 1.32 KiB {application} {hello_react} {server_rendering} [built]
[./node_modules/react/index.js] 189 bytes {application} {hello_react} {server_rendering} [built]
[./node_modules/react_ujs/react_ujs/index.js] 6.34 KiB {application} {server_rendering} [built]
    + 32 hidden modules

WARNING in ./app/javascript/components/REACT_COMMENTS.md 1:0
Module parse failed: Unexpected character '#' (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> # `React Commenting System`
| 
| **Last Updated: 3/13/2021**
 @ ./app/javascript/components sync ^\.\/.*$ ./REACT_COMMENTS.md
 @ ./app/javascript/packs/application.js
```

I think this means it basically worked? I'm going to try restarting it with the prebuild and seeing if it loads first-time without any intervention now. 🤞

**Comment by jywarren** [comment]
OK and one more question is: once we start a new prebuild up that has `webpack-cli`, `webpack-cli/serve`, and `wepback-dev-server` installed, do we have to actually run `webpack-dev-server` or is it automatically run by `passenger start`? I'll check the logs thoroughly in GitPod to try to find out.

**Comment by noi5e** [comment]
> I think that may have worked...? that is, after that, i can refresh the preview browser and it loads the site. I even caught the log saying something like Webpack compiling... and then printing the following:

That looks like a normal read-out for webpacker!

> But, did you always run webpack-dev-server or bin/webpack serve or something similar when you were developing locally?

I never did, webpacker always worked right out of the box after I ran `passenger start` and visited `localhost:3000`

**Comment by noi5e** [comment]
Barnstar error:
Screen Shot 2021-09-20 at 4 02 14 PM

**Comment by jywarren** [comment]
Hmm, yeah ...

```
gitpod /workspace/plots2 $ bin/webpack serve
yarn run v1.22.4
error Command "webpack" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

so i think i did install it but am not inserting the correct `serve` command?

**Comment by jywarren** [comment]
And now it's resulting in:

```

[Webpacker] Compiling...
[Webpacker] Compiling...
[Webpacker] Compilation failed:
yarn run v1.22.4
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


error Command "webpack" not found.

Completed 500 Internal Server Error in 10698ms (ActiveRecord: 59.1ms)


Webpacker can't find application.js in /workspace/plots2/public/packs/manifest.json. Possible causes:
1. You want to set webpacker.yml value of compile to true for your environment
```

so let me see again how to run `serve`

**Comment by jywarren** [comment]
Huh, this is interesting -- in `bin/` are 2 files, `webpack` and `webpack-dev-server`, both are ruby files!

image

I think the only difference is:

`require "webpacker/webpack_runner"` vs `require "webpacker/dev_server_runner"`

**Comment by jywarren** [comment]
Slight clue in latest error:

```
4. Your webpack configuration is not creating a manifest.
Your manifest contains:
{
}
 excluded from capture: Not configured to send/capture in environment 'development'
```

**Comment by jywarren** [comment]
i realized the npm module `webpack` wasn't installed, so added that, but still got the above error message

**Comment by jywarren** [comment]
Oof, now i can't get the app to serve without the manifest.json error!

**Comment by jywarren** [comment]
maybe in part because `public/packs/manifest.json` still doesn't exist?

**Comment by jywarren** [comment]
@noi5e what command actually generated your `manifest.json` file, regardless of which directory it was in? 

https://github.com/publiclab/plots2/pull/10137/#issuecomment-923406635 is actually further than i'm getting right now -- it actually had a manifest.json file, apparently.

**Comment by jywarren** [comment]
also learning: https://webpack.js.org/configuration/dev-server/

**Comment by noi5e** [comment]
I'm getting a little lost, I think I'm still behind where you're at in terms of understanding the issue.

Just to reiterate where I think we are right now:
1. The goal of this PR is to remove the `rails g webpacker:install && rails g webpacker:install:react && rails g react:install` from GitPod config.
2. The issue that we're dealing with now is that once those commands are removed, GitPod doesn't start. Instead, we get an empty `manifest.json` error.

> @noi5e what command actually generated your manifest.json file, regardless of which directory it was in?

I think that's done in the [install:react generator](https://github.com/reactjs/react-rails/blob/5f8723d6ac93b1c7899d40daa443f5315f78edcd/lib/generators/react/install_generator.rb#L71-L73)

I haven't logged as much time as you working on this issue, but my gut feeling is that the solution will probably be tweaking the `config/webpacker.yml` file's `development` settings. I think the webpacker docs will probably help.

I can also start drafting an issue for the `react-rails` people, but I think I need to understand what's going on with more clarity first.

**Comment by jywarren** [comment]
yeah i'm sorry, i'm a bit lost myself, it's so hard to keep an eye on the goal in the weeds like this. 

But, i just realized one thing - i've been adding extra modules via yarn to try to get this to work. But, i think i was getting a webpack version mismatch because i manually installed webpack via yarn! I tried uninstalling webpack via yarn and the site loaded correctly - i think my manual install was overriding the in-built version of webpack that comes in the rails/webpack gem!

So, i may need to look at removing the webpack-cli stuff too? and that `bin/webpack serve` stuff was a red herring - it should already be built in and managed by the rails/webpack gem.

**Comment by noi5e** [comment]
> that bin/webpack serve stuff was a red herring - it should already be built in and managed by the rails/webpack gem.

I think that's right! I've been browsing issues and solutions for this kind of thing (and we're not the only people dealing with this config stuff), and it seems like the solutions are usually more about tweaking lines in a config file.

> i think i was getting a webpack version mismatch because i manually installed webpack via yarn! I tried uninstalling webpack via yarn and the site loaded correctly - i think my manual install was overriding the in-built version of webpack that comes in the rails/webpack gem!

In reading through these issues, I have also noticed that issues can crop up because of how these packages are installed.

**Comment by jywarren** [comment]
OK, i'll try running this on a freshly prebuilt gitpod...

**Comment by jywarren** [comment]
I think @noi5e is right and we will have to also include a `rails g react:install` run. But let's see what happens now.

**Comment by jywarren** [comment]
Yes!!! `rails g react:install` makes it work, after initially seeing an error. Adding that and running prebuild once more...

**Comment by jywarren** [comment]
output was:

```
gitpod /workspace/plots2 $ rails g react:install
W, [2021-09-21T00:07:48.065496 #7847]  WARN -- Skylight: [SKYLIGHT] [5.1.1] Running Skylight in development mode. No data will be reported until you deploy your app.
(To disable this message for all local apps, run `skylight disable_dev_warning`.)
       exist  app/javascript/components
   identical  app/javascript/components/.keep
warning Pattern ["file-saver@eligrey/FileSaver.js#1.3.8"] is trying to unpack in the same destination "/home/gitpod/.cache/yarn/v6/npm-file-saver-1.3.8/node_modules/file-saver" as pattern ["file-saver@^1.3.3"]. This could result in non-deterministic behavior, skipping.
warning " > @babel/preset-react@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-display-name@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-jsx@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-jsx-development@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-pure-annotations@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/preset-react > @babel/plugin-transform-react-jsx > @babel/plugin-syntax-jsx@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > leaflet-spin@1.1.0" has incorrect peer dependency "leaflet@^0.7.3".
warning " > webpack-dev-server@4.2.1" has unmet peer dependency "webpack@^4.37.0 || ^5.0.0".
warning "webpack-dev-server > webpack-dev-middleware@5.1.0" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
warning Ignored scripts due to flag.                                   
File unchanged! The supplied flag value not found!  app/javascript/packs/application.js
   identical  app/javascript/packs/server_rendering.js
```

but, despite the warnings and "file unchanged" it seems to do the trick.

**Comment by jywarren** [comment]
Perfect. This worked. Merging!!

**Pull request merged**